File: IHaveATheory/app/src/main/java/com/avoid/ihaveatheory/game/choice/Choice.java
package com.avoid.ihaveatheory.game.choice;

import android.os.Parcel;
import android.os.Parcelable;

public abstract class Choice implements Parcelable{

    private String situation;
    private String choiceA;
    private String choiceB;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(situation);
        dest.writeString(choiceA);
        dest.writeString(choiceB);
    }

    public String getSituation() {
        return situation;
    }

    public void setSituation(String situation) {
        this.situation = situation;
    }

    public String getChoiceA() {
        return choiceA;
    }

    public void setChoiceA(String choiceA) {
        this.choiceA = choiceA;
    }

    public String getChoiceB() {
        return choiceB;
    }

    public void setChoiceB(String choiceB) {
        this.choiceB = choiceB;
    }

    public abstract void triggerChoiceA();

    public abstract void triggerChoiceB();
}

File: app/src/main/java/com/avoid/ihaveatheory/game/choice/Choice.kt
package com.avoid.ihaveatheory.game.choice

import android.os.Parcel
import android.os.Parcelable

abstract class Choice : Parcelable {

    var situation: String? = null
    var choiceA: String? = null
    var choiceB: String? = null

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(situation)
        dest.writeString(choiceA)
        dest.writeString(choiceB)
    }

    abstract fun triggerChoiceA()

    abstract fun triggerChoiceB()
}
